src/Generating/EnderDragonFightStructuresGen.h

```diff
@@ -0,0 +1,53 @@
+
+#pragma once
+
+/*
+https://minecraft.gamepedia.com/End_spike
+*/
+
+#include "FinishGen.h"
+
+class cEnderDragonFightStructuresGen :
+	public cFinishGen
+{
+public:
+	cEnderDragonFightStructuresGen(int a_Seed);
+
+protected:
+	cNoise m_Noise;
+	std::map<cChunkCoords, Vector3i> m_TowerPos;
+	struct sTowerProperties
+	{
+		int m_Height;
+		int m_Radius;
+		bool m_HasCage;
+
+		sTowerProperties(int a_Height, int a_Radius, bool a_HasCage) :
+			m_Height(a_Height),
+			m_Radius(a_Radius),
+			m_HasCage(a_HasCage)
+		{
+		}
+	};
+	std::array<sTowerProperties, 10> m_TowerHeights =
```

Comment:
Isn't it easier to determine these properties on the spot using an intnoise?
```cpp
int height = 70 + m_Noise.IntNoise3dInt(TowerPos) % 30;
int radius = 3 + m_Noise.IntNoise3dInt(TowerPos) % 3;
bool hasCage = (m_Noise.IntNoise3dInt(TowerPos) % 10) < 2;
```

Reply:
Yes it would be but the notchian implementation does have exakt value for that:
https://minecraft.gamepedia.com/End_spike

You clould do this on the fly if you like as there is a system behind the tower proportions

Reply:
I see. How about we make it configurable? Something like:
```ini
ObsydianPillars=76|3|false, 79|3|true, 82|3|true, ...
```

The code to generate the positions in a circle would have to be modified to handle different amounts of pillars though.

Reply:
That can be easily done. The code simply chooses an angle an starts placing towers evenly spaced in a circle. You can change the angle between towers to accommodate for more towers

Reply:
Do you think the radius of the circle should be modified or adjusted by the cound of towers

10 Towers -> 43 radius
20 Towers -> 86 radius
...

Reply:
You could make that configurable as well :)
```ini
ObsydianPillarsRadius=43
```